### PR TITLE
improve: Upgrade to sdk-v2 version with optimized caching logic

### PR DESCRIPTION
**Pull request by nicholaspai**
Should see speed ups in dataworker by:
- reducing time to cache utilization in HubPoolClient
- use redis cache for block finder getBlockForTimestamp functions

### CODE REVIEW

File: src/utils/BlockUtils.ts [hunk 1]
```diff
@@ -30,29 +31,5 @@ export async function getBlockForTimestamp(
...
-
-    // Expire key after 90 days.
-    if (shouldCache(timestamp, currentBlock.timestamp)) {
-      await setRedisKey(key, blockNumber.toString(), redisClient, 60 * 60 * 24 * 90);
-    }
-    return blockNumber;
-  } else {
-    return parseInt(result);
-  }
+  return getCachedBlockForTimestamp(chainId, timestamp, blockFinder ?? (await getBlockFinder(chainId)));
```
Comment (pxrl) [inline src/utils/BlockUtils.ts:34]: Not a showstopper, but resolving `blockFinder` this way is a bit dense.

**Review by pxrl — APPROVED** [review]
Looks OK to me, pending others' comments.

File: src/clients/HubPoolClient.ts [hunk 2]
```diff
@@ -12,11 +12,13 @@ export class HubPoolClient extends clients.HubPoolClient {
     configStoreClient: clients.AcrossConfigStoreClient,
     deploymentBlock?: number,
     chainId = 1,
-    eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 }
+    eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 },
+    timeToCache = 60 * 60
```
Comment (mrice32) [inline src/clients/HubPoolClient.ts:16]:
```suggestion
    timeToCache?: number
```

Why not just let this be undefined and use the default from the superclass?

File: src/utils/BlockUtils.ts (hunk 1 above)
Comment (mrice32) [reply src/utils/BlockUtils.ts]: Agreed, I would put the blockfinder resolution on a separate line above.

Comment (james-a-morris) [inline src/utils/BlockUtils.ts:34]:
```suggestion
  return utils.getCachedBlockForTimestamp(chainId, timestamp, blockFinder ?? (await getBlockFinder(chainId)));
```

File: src/utils/BlockUtils.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { BlockFinder, getProvider, getRedis, isDefined, setRedisKey, shouldCache } from "./";
+import { BlockFinder, getProvider, isDefined } from "./";
+import { getCachedBlockForTimestamp } from "@across-protocol/sdk-v2";
```
Comment (james-a-morris) [inline src/utils/BlockUtils.ts]:
```suggestion
import { utils } from "@across-protocol/sdk-v2";
```

**Review by mrice32 — APPROVED** [review]
Looks good!

File: src/common/Config.ts [hunk 4]
```diff
@@ -36,10 +37,16 @@ export class CommonConfig {
       SPOKE_POOL_CHAINS_OVERRIDE,
       ACROSS_BOT_VERSION,
       ACROSS_MAX_CONFIG_VERSION,
+      HUB_POOL_TIME_TO_CACHE,
     } = env;
 
     this.version = ACROSS_BOT_VERSION ?? "unknown";
 
+    this.timeToCache = Number(HUB_POOL_TIME_TO_CACHE ?? 60 * 60); // 1 hour by default.
+    if (this.timeToCache < 0) {
```
Comment (james-a-morris) [inline src/common/Config.ts:46]:
```suggestion
    if (Number.isNaN(this.timeToCache) || this.timeToCache < 0) {
```

Comment (nicholaspai) [reply src/common/Config.ts:46]: why do you think this is necessary? this seems redundant

Comment (james-a-morris) [reply src/common/Config.ts:46]: We should probably check for `Number.isNaN()` since `this.timeToCache < 0` fails if `this.timeToCache` is not a number

**Review by james-a-morris — APPROVED** [review]

**Pull request merged**